convert_values_test.go

```diff
@@ -30,209 +30,185 @@ import (
 	"github.com/onflow/cadence/runtime/sema"
 )
 
-func TestConvertVoidValue(t *testing.T) {
-	value := convertValue(interpreter.VoidValue{}, nil)
-
-	assert.Equal(t, NewVoid(), value)
-}
-
-func TestConvertNilValue(t *testing.T) {
-
-	value := convertValue(interpreter.NilValue{}, nil)
-
-	assert.Equal(t, NewOptional(nil), value)
-}
-
-func TestConvertSomeValue(t *testing.T) {
-	t.Run("Nil", func(t *testing.T) {
-		value := convertValue(&interpreter.SomeValue{Value: nil}, nil)
-
-		assert.Equal(t, NewOptional(nil), value)
-	})
-
-	t.Run("Non-nil", func(t *testing.T) {
-		value := convertValue(
-			&interpreter.SomeValue{Value: interpreter.NewIntValueFromInt64(42)},
-			nil,
-		)
-
-		assert.Equal(t, NewOptional(NewInt(42)), value)
-	})
-}
-
-func TestConvertBoolValue(t *testing.T) {
-	t.Run("True", func(t *testing.T) {
-		value := convertValue(interpreter.BoolValue(true), nil)
-
-		assert.Equal(t, NewBool(true), value)
-	})
-
-	t.Run("False", func(t *testing.T) {
-		value := convertValue(interpreter.BoolValue(false), nil)
-
-		assert.Equal(t, NewBool(false), value)
-	})
-}
-
-func TestConvertStringValue(t *testing.T) {
-	t.Run("Empty", func(t *testing.T) {
-		value := convertValue(&interpreter.StringValue{Str: ""}, nil)
-
-		assert.Equal(t, NewString(""), value)
-	})
-
-	t.Run("Non-empty", func(t *testing.T) {
-		value := convertValue(&interpreter.StringValue{Str: "foo"}, nil)
-
-		assert.Equal(t, NewString("foo"), value)
-	})
-}
-
-func TestConvertArrayValue(t *testing.T) {
-	t.Run("Empty", func(t *testing.T) {
-		value := convertValue(&interpreter.ArrayValue{Values: nil}, nil)
-
-		assert.Equal(t, NewArray([]Value{}), value)
-	})
-
-	t.Run("Non-empty", func(t *testing.T) {
-		value := convertValue(
-			&interpreter.ArrayValue{
-				Values: []interpreter.Value{
-					interpreter.NewIntValueFromInt64(42),
-					interpreter.NewStringValue("foo"),
-				},
+var convertTests = []struct {
+	label       string
+	value       interpreter.Value
+	expected    Value
+	skipReverse bool
+}{
+	{
+		label:    "Void",
+		value:    interpreter.VoidValue{},
+		expected: NewVoid(),
+	},
+	{
+		label:       "Nil",
+		value:       interpreter.NilValue{},
+		expected:    NewOptional(nil),
+		skipReverse: true,
+	},
+	{
+		label:    "SomeValue nil",
+		value:    &interpreter.SomeValue{Value: nil},
+		expected: NewOptional(nil),
+	},
```

Comment:
The case "`interpreter.SomeValue` with `nil`" doesn't really exist, it's basically the "Nil" case above, i.e. Cadence `nil` is represented as `interpreter.NilValue`.

Maybe introduce a `cadence.Nil` instead of `cadence.Optional{nil}`, or remove this conversion/test case here and don't skip reverse for the "Nil" case above